cmd/vic-machine/validate/validator.go

```diff
@@ -45,13 +48,183 @@ type Validator struct {
 
 	Session *session.Session
 	Context context.Context
+
+	issues []error
 }
 
 func NewValidator() *Validator {
 	return &Validator{}
 }
 
-func (v *Validator) Validate(input *data.Data) (*metadata.VirtualContainerHostConfigSpec, error) {
+func (v *Validator) NoteIssue(err error) {
+	if err != nil {
+		v.issues = append(v.issues, err)
+	}
+}
+
+func (v *Validator) ListIssues() error {
+	if len(v.issues) == 0 {
+		return nil
+	}
+
+	for _, err := range v.issues {
+		fmt.Println(err)
+	}
+
+	return errors.New("Validation of configuration failed")
+}
+
+func (v *Validator) Validate2(ctx context.Context, input *Data, conf *metadata.VirtualContainerHostConfigSpec) (*metadata.VirtualContainerHostConfigSpec, error) {
+	var targetURL url.URL
+	targetURL.Host = input.target
+	targetURL.Path = "sdk"
+	targetURL.User = url.UserPassword(input.user, *input.passwd)
+
+	conf.Target = targetURL
+	conf.Insecure = input.insecure
+
+	// TODO: ensure that displayname doesn't violate constraints (length, characters, etc)
+	conf.SetName(input.displayName)
+
+	// Compute
+	pool, err := v.ResourcePool(ctx, input.computeResourcePath)
+	v.NoteIssue(err)
+	conf.AddComputeResource(pool)
+
+	// Image Store
+	ds, err := v.DatastorePath(ctx, input.imageDatastoreName)
+	v.NoteIssue(err)
+	conf.AddImageStore(ds)
+
+	// TODO: add volume locations
+
+	// External net
+	extMoref, err := v.Network(ctx, input.externalNetworkName)
+	v.NoteIssue(err)
+	conf.AddNetwork(&metadata.NetworkEndpoint{
+		Network: metadata.ContainerNetwork{
+			Common: metadata.Common{
+				Name: "external",
+				ID:   fmt.Sprintf("%s-%s", extMoref.Type, extMoref.Value),
+			},
+		},
+	})
+
+	// Bridge net
+	bridgeMoref, err := v.Network(ctx, input.bridgeNetworkName)
+	v.NoteIssue(err)
+	conf.AddNetwork(&metadata.NetworkEndpoint{
+		Network: metadata.ContainerNetwork{
+			Common: metadata.Common{
+				Name: "bridge",
+				ID:   fmt.Sprintf("%s-%s", bridgeMoref.Type, bridgeMoref.Value),
+			},
+		},
+	})
+
+	// Management net
+	if input.managementNetworkName != "" {
+		managementMoref, err := v.Network(ctx, input.managementNetworkName)
+		v.NoteIssue(err)
+		conf.AddNetwork(&metadata.NetworkEndpoint{
+			Network: metadata.ContainerNetwork{
+				Common: metadata.Common{
+					Name: "bridge",
+					ID:   fmt.Sprintf("%s-%s", managementMoref.Type, managementMoref.Value),
+				},
+			},
+		})
+	}
+
+	// TODO: add client network
+
+	// TODO: add mapped networks
+
+	// Perform the higher level compatibility and consistency checks
+	err = v.CompatibilityChecks(ctx, conf)
+	v.NoteIssue(err)
+
+	// TODO: determine if this is where we should turn the noted issues into message
+	return conf, v.ListIssues()
+}
+
+func (v *Validator) Network(ctx context.Context, path string) (*types.ManagedObjectReference, error) {
+	nets, err := v.Session.Finder.NetworkList(ctx, path)
```

Comment:
Why not just use Finder.Network()? It will return error message for not found and multiple found.